kube-client/src/client/builder.rs

```diff
@@ -101,8 +100,11 @@ impl TryFrom<Config> for ClientBuilder<BoxService<Request<hyper::Body>, Response
             ));
 
             let mut connector = TimeoutConnector::new(connector);
-            connector.set_connect_timeout(timeout);
-            connector.set_read_timeout(timeout);
+
+            // Set the timeout for the client and fallback to default deprecated timeout until it's removed
+            connector.set_connect_timeout(config.connect_timeout.or(config.timeout));
```

Comment:
Will this warn the users as well? We might need to add `#[allow(deprecated)]`.

It might also be fine to just remove it. I don't think it's configured often, and the fix should be trivial.

Reply:
For what it's worth, warnings are silenced for upstream dependencies IIRC, this should only be annoying for us.

Reply:
I also considered simply removing it instead of a deprecation as the replacement is fairly straightforward. 

But as a consumer the packages I like having a deprecation warning before something is removed.

Up to you, I'm fine with either way.

Reply:
It will show up on every PR here until we remove the field though.

I would also want to add `#[allow(deprecated)]` on our internal usage, it's meant to notify users rather than every PR for the next 5 months with 5 warnings (as you can see are inlined in the diff).

Reply:
Yeah, true.

Reply:
Done! Please check if I did it correctly, my first time using it.

Reply:
Looks good, not seeing CI show any warnings anymore. Thanks!